dht/range_streamer.cc

```diff
@@ -301,9 +301,6 @@ future<> range_streamer::stream_async() {
                         do_streaming();
                     }
                 } catch (...) {
-                    for (auto& range : ranges_to_stream) {
-                        range_vec.push_back(range);
-                    }
```

Comment:
Eventually this is part of _to_stream. How do you know it's not reused?

Reply:
@asias

Reply:
> Eventually this is part of _to_stream. How do you know it's not reused?

I inspected all call sites

Reply:
We will report wrong nr_ranges_remaining. After this patch, the failed ranges will be reported as ranges finished successfully.

Reply:
The commit log says:

```
    dht/range_streamer: stream_async no need to return ranges to range_vec on exception
    
    There is no retry when stream_async fails
    so no need to return the failed ranges to range_vec.
    
    This will make it easier to move ranges_to_stream
    does the stack rather than copying it in the following
    patches.
```
 
However, I think range_streamer::stream_async() should keep the failed ranges instead of dropping it silently, we do not retry in stream_async (any more) does not mean the caller of the stream_async() woulc not do the retry it self.